tests/src/test/scala-3/pureconfig/CoproductReaderDerivationSuite.scala

```diff
@@ -0,0 +1,37 @@
+package pureconfig
+
+import scala.language.higherKinds
+
+import com.typesafe.config.ConfigFactory
+import pureconfig._
+import pureconfig.error._
+import pureconfig.generic._
+import pureconfig.generic.derivation.default.derived
+
+enum AnimalConfig derives ConfigReader {
+  case DogConfig(age: Int)
+  case CatConfig(age: Int)
+  case BirdConfig(canFly: Boolean)
+}
+
+class CoproductReaderDerivationSuite extends BaseSuite {
```

Comment:
We should also add one test for when the hint field exists but it doesn't correspond to an existing option.